apis/keda/v1alpha1/scaledobject_webhook.go

```diff
@@ -255,19 +255,18 @@ func verifyCPUMemoryScalers(incomingSo *ScaledObject, action string) error {
 				if conainerName != "" && container.Name != conainerName {
 					continue
 				}
+
 				if trigger.Type == cpuString {
-					if container.Resources.Requests == nil ||
-						container.Resources.Requests.Cpu() == nil ||
-						container.Resources.Requests.Cpu().AsApproximateFloat64() == 0 {
+					if container.Resources.Requests.Cpu().AsApproximateFloat64() == 0 &&
+						container.Resources.Limits.Cpu().AsApproximateFloat64() == 0 {
```

Comment:
Are you sure that this won't raise a panic at some time? We are using the deployment spec and not the pod spec, so Request can be `nil` (and the same for the limits). Shouldn't we check `nil` before dereferencing the value?

Reply:
Better safer than sorry, isn't?

I try to simplify every code I touch, but in this case, you're right, I cannot assume that Kubernetes internal methods will never return `nil`. Following Murphy's Law it would break in the next Kubernetes SDK release. 😛 

Sorry. 😥 I'll rewrite it.